src/textchannellistener.cpp

```diff
@@ -690,13 +682,11 @@ void TextChannelListener::handleMessages()
 
             // class 0 sms
             if (m_isClassZeroSMS) {
-                // just ack message, expunge would be called
-                // as soon as user reads message
-                DEBUG() << __FUNCTION__ << "Adding class 0 sms";
+                DEBUG() << __FUNCTION__ << "Handling class 0 sms";
                 processedMessages << message;
-                classZeroSMSModel()->addEvent(event,true);
-                m_EventTokens.insertMulti(event.id(), event.messageToken());
                 nManager->playClass0SMSAlert();
+                nManager->requestClass0Notification(event);
+                expungeMessage(event.messageToken());
```

Comment:
expungeMessage() looks suspicious, btw. Sets expunge timer only if something was already added to m_expungeTokens. Am I missing something?